rectools/metrics/debias.py

```diff
@@ -0,0 +1,162 @@
+#  Copyright 2024 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Debias module."""
+
+import typing as tp
+
+import attr
+import pandas as pd
+
+from rectools import Columns
+
+from .base import MetricAtK
+
+
+@attr.s(frozen=True)
+class DebiasConfig:
+    """
+    Config with debias method parameters.
+
+    Parameters
+    ----------
+    iqr_coef : float, default 1.5
+        Coefficient for defining as the maximum value inside the border.
+    random_state : int, optional, default None
+        Pseudorandom number generator state to control the down-sampling.
+    """
+
+    iqr_coef: float = attr.ib(default=1.5)
+    random_state: tp.Optional[int] = attr.ib(default=None)
+
+
+@attr.s
+class DebiasableMetrikAtK(MetricAtK):
+    """
+    Classification metric base class.
+
+    Warning: This class should not be used directly.
+    Use derived classes instead.
+
+    Parameters
+    ----------
+    k : int
+        Number of items at the top of recommendations list that will be used to calculate metric.
+    debias_config : DebiasConfig, default None
+        Config with debias method parameters (iqr_coef, random_state).
+    """
+
+    debias_config: DebiasConfig = attr.ib(default=None)
+
+    def _check_debias(self, is_debiased: bool, type_of_intermediate_values_of_interactions: str) -> None:
+        if not is_debiased and self.debias_config is not None:
+            raise ValueError(
+                "You have specified `debias_config` for metric "
+                f"but `{type_of_intermediate_values_of_interactions}` is not assumed to be de-biased. "
```

Comment:
Current version sounds like "this thing should not be debiased"
```suggestion
                f"but `{type_of_intermediate_values_of_interactions}` is not de-biased. "
```